### PR TITLE
style: add typescript/jsdoc types

### LINKED ISSUE #3: Add TypeScript doc files
Example from Expo

`build.d.ts`
```
declare enum BackgroundFetchResult {
    NoData = 1,
    NewData = 2,
    Failed = 3
}
declare enum BackgroundFetchStatus {
    Denied = 1,
    Restricted = 2,
    Available = 3
}
interface BackgroundFetchOptions {
    minimumInterval?: number;
    stopOnTerminate?: boolean;
    startOnBoot?: boolean;
}
export declare function getStatusAsync(): Promise<BackgroundFetchStatus | null>;
export declare function setMinimumIntervalAsync(minimumInterval: number): Promise<void>;
export declare function registerTaskAsync(taskName: string, options?: BackgroundFetchOptions): Promise<void>;
export declare function unregisterTaskAsync(taskName: string): Promise<void>;
export { BackgroundFetchResult as Result, BackgroundFetchStatus as Status, };
```

### LINKED ISSUE #16: Importing "index.js" in browser does not work
By importing

```
import deepEqual from './utils/deepEqual';
```

The import does not work in browser, because the extension is not set.

Use instead:
```
import deepEqual from './utils/deepEqual.js';
```

### PR DESCRIPTION
**Pull request by jehon**
Fix #3 
Fix #16

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
- * }
+ * Manage a variable accross multiple files
+ * The data is associated to a "key" (can be anything)
  */
-let store = {};
-
 export default {
+  /**
+   * Store data into the storage
+   * @param {*} key: the key under wich the value will be stored
```
Comment (BrodaNoel) [inline index.js:11]: Shouldn't `key` be recommended as `string`?
We don't have any validation there (and I don't think we really need it), but recommending a `string` may be more helpful than `any`

File: index.js [hunk 2]
```diff
@@ -33,10 +31,22 @@ export default {
     store[key].subscribers.forEach(callback => callback(newValue, prevValue));
   },
 
+  /**
+   * Return the data associated to the key
+   * @param {*} key
```
Comment (BrodaNoel) [inline index.js:36]: String here as well

File: index.js [hunk 3]
```diff
@@ -33,10 +31,22 @@ export default {
...
+   */
   get(key) {
     return !store[key] ? undefined : store[key].value;
   },
 
+  /**
+   *
+   * @param {*} key - the key of the data
```
Comment (BrodaNoel) [inline index.js:44]: String here as well

**Review by BrodaNoel — COMMENTED** [review]
Tell me what do you think about the `string` for `key` (only in the documentation. not adding a validation).

File: index.js [hunk 4]
```diff
@@ -1,18 +1,16 @@
 import deepEqual from './utils/deepEqual';
```
Comment (BrodaNoel) [inline index.js]:
```suggestion
import deepEqual from './utils/deepEqual.js';
```

**Comment by jehon** [comment]
The key could also be a Symbol...
And saying it is a string in the comment, would imply that "2" === 2 by automatic translation, what it is not doing... So even if it would imply clarity in the doc, that would mislead some... So I would not say that

**Comment by jehon** [comment]
Fix #16

**Pull request merged**